packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeNodeRenderer.tsx

```diff
@@ -68,68 +73,42 @@ type TreeNodeRendererProps = Pick<ReturnType<typeof useTree>, "expandNode" | "is
  */
 export const TreeNodeRenderer: ForwardRefExoticComponent<TreeNodeRendererProps & RefAttributes<HTMLElement>> = forwardRef(
   (
-    {
-      node,
-      expandNode,
-      getLabel,
-      getSublabel,
-      onFilterClick,
-      onNodeClick,
-      onNodeKeyDown,
-      isNodeSelected,
-      getHierarchyLevelDetails,
-      reloadTree,
-      actions,
-      getDecorations,
-      ...treeItemProps
-    },
+    { node, expandNode, getLabel, getSublabel, onNodeClick, onNodeKeyDown, isNodeSelected, actions, getDecorations, hasError, reloadTree, ...treeItemProps },
     forwardedRef,
   ) => {
     const nodeRef = useRef<HTMLElement>(null);
     const ref = useMergedRefs(forwardedRef, nodeRef);
+    const { localizedStrings } = useLocalizationContext();
+
+    const childErrorNode = useMemo(() => {
```

Comment:
Do we need this `useMemo`? Without it you won't need to do additional check for placeholder node.

Reply:
I don't think we expect the node to change much after loading so I think having it as useMemo is good. Quick test with console.log seems to show that useMemo saves some calls.